dvc/commands/repro.py

```diff
@@ -117,10 +117,7 @@ def add_arguments(repro_parser):
         "--pull",
         action="store_true",
         default=False,
-        help=(
-            "Try automatically pulling missing cache for outputs restored "
-            "from the run-cache."
-        ),
+        help="Try automatically pulling missing data.",
```

Comment:
This message should have been updated before

Reply:
so, to confirm - it pulls only if stage is otherwise unchanged?

Reply:
It pulls the stage _outputs_ and skips running if the stage is otherwise unchanged.

The other part of `--pull` is to pull all [missing deps](https://github.com/iterative/dvc/blob/e82356ba9213394af5a9e91ac8026c3c09670bf9/dvc/stage/run.py#L119-L122), which happens when the stage runs.

Reply:
Overall, the goal is to pull all missing data needed without pulling unnecessary data. For example, if a param changed, we won't pull the outputs since the stage needs to re-run and overwrite the outputs anyway.

Reply:
okay, makes sense, probably answers my second question here https://github.com/iterative/dvc/pull/10435/files#r1610266807

Reply:
for whatever reason it always takes some time for me to wrap my mind around --alow-missing and --pull, etc. I wonder if we can explain them a bit more explicit with examples like you did in your comment above